ForumSessionProvider/ForumSessionProvider.php

```diff
@@ -14,288 +14,759 @@
+	protected $userWiki;
+
+	// Other Variables.
+	protected $loaded = false;
+	protected $logger;
+
+	/*
+	 * Starts our session handler.  All the work starts here.
+	*/
+	public function __construct(array $params = array())
+	{
+		global $wgSMFLogin, $wgFSPSoftware;
+
 		parent::__construct($params);
 
 		// TODO: Keep eye out for other things that need logging. Admin changes?
 		$this->logger = LoggerFactory::getInstance('ForumSessionProviderLog');
 
-        // Administrator always get admin rights.
-        if (!in_array(1, $GLOBALS['wgFSPAdminGroups']))
-            $GLOBALS['wgFSPAdminGroups'][] = 1;
+		// Set our software up.
+		$this->wgFSPSoftware = !empty($wgSMFLogin) ? 'smf2.0' : (!empty($wgFSPSoftware) ? $wgFSPSoftware : null);
+		$wgFSPSoftware = $this->wgFSPSoftware;
 
-        if (!in_array(1, $GLOBALS['wgFSPInterfaceGroups']))
-            $GLOBALS['wgFSPInterfaceGroups'][] = 1;
-		
-		if (is_readable($GLOBALS['wgFSPPath'] . '/Settings.php')) {
-		    require ($GLOBALS['wgFSPPath'] . '/Settings.php');
-		    // Globals declared in Settings.php gain local scope.
-		    $GLOBALS['wgFSPCookieName'] = $cookiename;
-		    $GLOBALS['wgFSPBoardURL'] = $boardurl;
+		// Load our settings.
+		$this->loadFSSettings();
+
+		// This is old Auth_SMF.php extension? Lets compat it.
+		if (!empty($wgSMFLogin) || in_array($wgFSPSoftware, array('smf2.0', 'smf2.1')))
+			$this->compatAuthSMF();
+
+		// Make sure we can find the settings file.
+		if (
+			!empty($this->FSSettings['path'])
+			&& is_readable($this->FSSettings['path'] . '/Settings.php')
+		)
+		{
+			// Read the Settings file in, use this layer to adjust what we need to bring in.
+			$this->readFSFSettings($this->FSSettings['path'] . '/Settings.php');
 
+			// Read the cookie
 			$this->decodeCookie();
 
-			if ($this->id && is_integer($this->id)) {
-                $this->prefix = $db_prefix;
+			// If we have a valid ID, lets connect to the database.
+			if ($this->FScookie['id'] && is_integer($this->FScookie['id']))
+				$this->FSDBConnect();
+		}
+		else
+			$this->logger->warning('Settings.php missing or not readable. Tried to load: ' . $this->FSSettings['path'] . '/Settings.php');
+	}
+
+	/*
+	 * Load up all Forum Software Settings using the standard way.
+	*/
+	public function loadFSSettings()
+	{
+		global $wgFSPPath, $wgFSDebug, $wgFSPDenyGroups, $wgFSPAllowGroups, $wgFSPAdminGroups, $wgFSPSuperGroups, $wgFSPSpecialGroups, $wgFSPInterfaceGroups, $wgFSPEnableBanCheck;
+
+		$this->FSSettings['path'] = isset($wgFSPPath) ? $wgFSPPath : '../forum';
+		$this->FSSettings['debug'] = !empty($wgFSDebug) ? true : false;
+
+		// Some standard groups.
+		foreach (array(
+			'LoginDeniedGroups' => 'wgFSPDenyGroups',
+			'LoginAllowedGroups' => 'wgFSPAllowGroups',
+			'AdminGroups' => 'wgFSPAdminGroups',
+			'SuperAdminGroups' => 'wgFSPSuperGroups',
+			'InterfaceGroups' => 'wgFSPInterfaceGroups',
+			'SpecialGroups' => 'wgFSPSpecialGroups',
+			'NameStyle' => 'wgFSPNameStyle',

```

Comment:
This isn't getting properly read. Nor is it defaulting properly when it fails to read it.